tests/MongoDB.Driver.Examples/GenerateKeyExamples.cs

```diff
@@ -0,0 +1,261 @@
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using MongoDB.Bson;
+using MongoDB.Driver.Encryption;
+using Xunit;
+
+namespace MongoDB.Driver.Examples
+{
+    public class GenerateKeyExamples
+    {
+        // https://gist.github.com/DmitryLukyanov/8417d1f5c30f12ffeb2be29056aff455#file-localkmsprovider-cs
+        [Fact]
+        public void LocalKmsProvider()
+        {
+            // 2. Code full example: generate a 96-byte master key and save to a file called master-key.txt.
+            string localMasterKeyBase64;
+            using (var randomNumberGenerator = System.Security.Cryptography.RandomNumberGenerator.Create())
+            {
+                var bytes = new byte[96];
+                randomNumberGenerator.GetBytes(bytes);
+                localMasterKeyBase64 = Convert.ToBase64String(bytes);
+                Console.WriteLine(localMasterKeyBase64);
+                File.WriteAllText("master-key.txt", localMasterKeyBase64);
+            }
+
+            // 3. generate a data key using the master key read from the file master-key.txt
+            localMasterKeyBase64 = File.ReadAllText("master-key.txt");
+            var localMasterKeyBytes = Convert.FromBase64String(localMasterKeyBase64);
+
+            var kmsProviders = new Dictionary<string, IReadOnlyDictionary<string, object>>();
+            var localOptions = new Dictionary<string, object>
+            {
+                { "key", localMasterKeyBytes }
+            };
+            kmsProviders.Add("local", localOptions);
+
+            var connectionString = "mongodb://localhost:27017";
+            var keyVaultNamespace = CollectionNamespace.FromFullName("encryption.__keyVault");
+            var keyVaultClient = new MongoClient(connectionString);
+            var clientEncryptionOptions = new ClientEncryptionOptions(
+                keyVaultClient: keyVaultClient,
+                keyVaultNamespace: keyVaultNamespace,
+                kmsProviders: kmsProviders);
+
+            var clientEncryption = new ClientEncryption(clientEncryptionOptions);
+            var dataKeyId = clientEncryption.CreateDataKey("local", new DataKeyOptions(), CancellationToken.None);
+            Console.WriteLine($"DataKeyId [UUID]: {dataKeyId}");
+            var dataKeyIdBase64 = Convert.ToBase64String(GuidConverter.ToBytes(dataKeyId, GuidRepresentation.Standard));
+            Console.WriteLine($"DataKeyId [base64]: {dataKeyIdBase64}");
+
+            // Validate key
```

Comment:
Let's have a more precise comment:

    // verify that the key document was created

We're not "validating the key". That would involve examining the contents of the key document to see if they were "valid" whatever that means.

Reply:
Also... what's the point of lines 68-83?

Are we implying that our users can't trust the call to `CreateDataKey` and that they must always verify that the key document was actually created?